src/transactions/reducer.ts

```diff
@@ -145,12 +145,65 @@ export const reducer = (
         }
       })
 
+      const receivedTransactions: TokenTransaction[] = []
+      const pendingTxsWithStandbyMatch: TokenTransaction[] = []
+      action.transactions.forEach((tx) => {
+        if (
+          tx.status === TransactionStatus.Pending &&
+          state.standbyTransactions.find(
+            (standbyTx) =>
+              tx.transactionHash === standbyTx.transactionHash &&
+              tx.networkId === standbyTx.networkId
+          )
+        ) {
+          pendingTxsWithStandbyMatch.push(tx)
+        } else {
+          receivedTransactions.push(tx)
+        }
+      })
+
```

Comment:
Maybe we should add a comment of what this logic is trying to do? and why?
What's in the PR description.
Or I guess the tests are good too, and we could point the reader to them?